drivers/unix/dir_access_unix.cpp

```diff
@@ -211,36 +211,38 @@ Error DirAccessUnix::make_dir(String p_dir) {
 Error DirAccessUnix::change_dir(String p_dir) {
 
 	GLOBAL_LOCK_FUNCTION
-	p_dir = fix_path(p_dir);
 
-	char real_current_dir_name[2048];
-	getcwd(real_current_dir_name, 2048);
-	String prev_dir;
-	if (prev_dir.parse_utf8(real_current_dir_name))
-		prev_dir = real_current_dir_name; //no utf8, maybe latin?
+	// make sure current_dir is valid absolute path
+	if (current_dir == "." || current_dir == "") {
+		char real_current_dir_name[2048];
+		getcwd(real_current_dir_name, 2048);
+		current_dir.parse_utf8(real_current_dir_name);
+	}
```

Comment:
There are problems with these changes. Now `change_dir` no longer resets the current working directory to its original state. The variable that is used to reset it is `prev_dir`, which must always be the directory resulted from `getcwd` before any of the `chdir` calls. With this commit, that only happens if the condition `current_dir == "." || current_dir == ""` is true; otherwise `prev_dir` is set to the directory the `DirAccessUnix` is at (`current_dir`).

For a quick test you can check the result of `getcwd` at the beginning and the end of this loop:

https://github.com/godotengine/godot/blob/ed606ded52132f7afed03d99fabed7b111ce9a51/core/project_settings.cpp#L331-L346